src/compiler/checker.ts

```diff
@@ -10585,6 +10585,9 @@ namespace ts {
 
         function elaborateError(node: Expression | undefined, source: Type, target: Type): boolean {
             if (!node || isOrHasGenericConditional(target)) return false;
+            if (!isTypeAssignableTo(source, target) && elaborateDidYouMeanToCallOrConstruct(node, source, target)) {

```

Comment:
This should ideally be cheap since the original call will have some of the relationships cached, right?

Also, technically I think you should be using the original relationship, not assignability, right?

Reply:
Yes and yes.